### DISCUSSION TITLE
Crud and loader/progressbar

### DISCUSSION THREAD

**Discussion by blondie63** [Laravel-Backpack/community-forum#906] [Q&A (Help)] [answered]
I've a bkp6/lv11 crud where i've a button that call a function that send emails 
This function take time so i'd like to show a popup with a kind of loader or progress bar
Is there a simply way to do it ?

Thanks for help

**Comment by pxpm** [comment]
Hey @blondie63 , Not now. 

We had in times a loader in Backpack. But we never used it and were not really happy about it's looks, so I think we removed it last version. 

Happy to brainstorm in some suggestions. I think the biggest problem we had at the time, is that sometimes the loader looks better as a full page loader, other times having a load in button was better (like we have in inline create button). 

We didn't find something we can generally use, so we always endup creating loaders to fit our needs. 

Maybe it's time to revisit that 🤷‍♂️ 

In the meanwhile you can check the source of Inline Create button, I think when you save the entry, it shows you a loading indicator in the button. 

Cheers

**Reply by blondie63** [reply]
do you mean this: CRUD::addButtonFromView('line', 'send_mail_users' , 'beginning'); ?

**Reply by karandatwani92** [reply, marked as answer]
Yes, you have to [create a custom button ](https://backpackforlaravel.com/docs/6.x/crud-buttons#creating-a-custom-button-1) for it and call the operation as Ajax requests.

You can take a look at delete button for reference `vendor/backpack/crud/src/resources/views/crud/buttons/delete.blade.php`
. This sends an Ajax request.

You can copy its definition and create a new button with the following:
```
php artisan backpack:button new-button-name --from=delete
```
Now, Modify it and use it per your need.

**Reply by blondie63** [reply]
@pxpm Do you like it ?


https://github.com/Laravel-Backpack/community-forum/assets/1261851/e0965733-3be0-46ce-8d92-acc4d3790a4f

**Reply by blondie63** [reply]
I've used Tailwindcss because i'm just using it in other blades but my previous developer import it in each blade with:
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/tailwindcss/2.2.19/tailwind.min.css">
Now i'd like to use basset to have last tailwind 3.4.3 version global and optimized
Can you advise me how to do it?

**Reply by karandatwani92** [reply]
Hey @blondie63 

It's another question not related to this discussion. I request you open another discussion if you are still seeking a solution.

From what I understand with the given details, you need `<script src="{{ basset('https://cdn.com/path/to/file.js') }}">`

**Reply by pxpm** [reply]
@blondie63 for example in the case you present I would had preferred some loader only in the table, or even only the button displaying a loading indicator since nothing changes in table. 

I think those full loaders are more appropriate when you are doing heavy changes on the background page, imagine that you removed the table and draw some Form or similar.

It's subjective, it's my opinion, and I think opinions vary a lot 🤷
